src/storage/lock_manager/deadlock.rs

```diff
@@ -464,7 +500,7 @@ impl Detector {
             return;
         }
 
-        let detect_table = Rc::clone(&self.inner.borrow().detect_table);
+        let inner = Rc::clone(&self.inner);
```

Comment:
Is it possible the inner is not `leader` here?

Reply:
No.  line 494 handles it.